### PR TITLE
RBTA-99/bug report/comments/general/volodymyr-cherniak

### PR DESCRIPTION
**Pull request by Volodymyr-Cherniak**
https://jira.pasv.us/browse/RBTA-99

Created tests for Bug-report Comments General

SPECK==> '.test/specs/reg/bug-report/comments.js'
SELECTORS==>'.test/selectors/bug-report.js'
EXPECTED==> '.test/expected/bug-report-comments.js'
HELPERS==> '.test/helpers/comments.js'

### CODE REVIEW

File: test/specs/reg/bug-report/comments.js [hunk 1]
```diff
@@ -38,3 +38,39 @@ describe('Comments Title', function () {
...
+
+describe('General', function () {
+
+    it('entered in bug-report form', function () {
+        help.login();
+        $('#to_me').click();
+        browser.pause(2000);
+        $('.text-left').waitForDisplayed(true);
+        $('.text-left').click();
+        let actual = $('#edit').isDisplayed();
```
[the remark this answers is not in the document]
Comment (Volodymyr-Cherniak) [reply test/specs/reg/bug-report/comments.js]: done

File: test/specs/reg/bug-report/comments.js [hunk 2]
```diff
@@ -38,3 +38,39 @@ describe('Comments Title', function () {
...
 
 });
+
+describe('General', function () {
+
+    it('entered in bug-report form', function () {
+        help.login();
+        $('#to_me').click();
+        browser.pause(2000);
+        $('.text-left').waitForDisplayed(true);
```
[the remark this answers is not in the document]
Comment (Volodymyr-Cherniak) [reply test/specs/reg/bug-report/comments.js]: done

File: test/helpers/comments.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import exp from "../expected/bug-report-comments";
+import sel from "../selectors/bug-report";
+
+class Helpers {
+
+    addNewComments(){
+        let addComment = $(sel.addNewComments);
+        addComment.addValue('Q'.repeat(5));
```
Comment (artsenius) [inline test/helpers/comments.js]: 1) you can pass '5' as an argument into the function.
2) also, you can add line breaks to make it multiline comment

File: test/selectors/bug-report.js [hunk 4]
```diff
@@ -16,5 +16,13 @@ export default {
   sections: ['#attachment', '#summary', '#str', '#actual', '#expected', '#assignee', '#repro', '#severity', '#priority', '#reporter', '#version'],
   titleText: '.text-bold',
   valueText: '.text-gray',
-  cancel: '#cancel'
+  cancel: '#cancel',
+  comments: '#comments',
+  commentsLength: '#comments div div div.text-gray',
+  commentsWritten: '#comments div div div div',
+  addNewComments: '.form-control',
+  buttonAddComments: '.btn.btn-primary.align-right',
```
Comment (artsenius) [inline test/selectors/bug-report.js]: there are ids you can use. Why did you choose tags instead?

File: test/specs/reg/bug-report/comments.js [hunk 5]
```diff
@@ -38,3 +39,51 @@ describe('Comments Title', function () {
   });
 
 });
+
+describe('General', function () {
+
+    it('entered in bug-report form', function () {
+        help.login();
+        $(sel.buttonAssignedToMe).click();
+        browser.pause(1000);
```
Comment (artsenius) [inline test/specs/reg/bug-report/comments.js]: too long. Try 200-300ms

File: test/specs/reg/bug-report/comments.js [hunk 6]
```diff
@@ -38,3 +39,51 @@ describe('Comments Title', function () {
...
+        $('.text-left').click();
```
Comment (artsenius) [inline test/specs/reg/bug-report/comments.js]: hardcoded. And why you use class? There are ids. You can click using '#summary_1 span' at least.

File: test/specs/reg/bug-report/comments.js [hunk 7]
```diff
@@ -38,3 +39,51 @@ describe('Comments Title', function () {
...
+        assert.equal(title, exp.commentsName);
+    });
+
+    it('New comment text area', function () {
+        let actual = $(sel.comments).isEnabled('form-control');
+        assert.equal(actual, true);
+    });
+
+    it('Module has "Add Comment” button', function () {
+        let actual = $(sel.comments).isEnabled('.btn.btn-primary.align-right');
```
Comment (artsenius) [inline test/specs/reg/bug-report/comments.js]: hardcoded values

File: test/specs/reg/bug-report/comments.js [hunk 8]
```diff
@@ -38,3 +39,51 @@ describe('Comments Title', function () {
...
+    });
+
+    it('Module has title “Comments”', function () {
+        let com = $(sel.commentsTitle);
+        let title = com.getText();
+        assert.equal(title, exp.commentsName);
+    });
+
+    it('New comment text area', function () {
+        let actual = $(sel.comments).isEnabled('form-control');
```
Comment (artsenius) [inline test/specs/reg/bug-report/comments.js]: hardcoded values

File: test/helpers/comments.js (hunk 3 above)
Comment (Volodymyr-Cherniak) [reply test/helpers/comments.js]: I can not do due to lack of time. In the future I will try to do

File: test/selectors/bug-report.js (hunk 4 above)
Comment (Volodymyr-Cherniak) [reply test/selectors/bug-report.js]: done

File: test/specs/reg/bug-report/comments.js (hunk 5 above)
Comment (Volodymyr-Cherniak) [reply test/specs/reg/bug-report/comments.js]: I changed the test, all works without "browser.pause()"

File: test/specs/reg/bug-report/comments.js (hunk 6 above)
Comment (Volodymyr-Cherniak) [reply test/specs/reg/bug-report/comments.js]: done

File: test/specs/reg/bug-report/comments.js (hunk 7 above)
Comment (Volodymyr-Cherniak) [reply test/specs/reg/bug-report/comments.js]: done

File: test/specs/reg/bug-report/comments.js (hunk 8 above)
Comment (Volodymyr-Cherniak) [reply test/specs/reg/bug-report/comments.js]: done

**Pull request merged**